hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -986,6 +992,15 @@ CompletableFuture<Void> gracefullyTerminate() {
         return future.thenCompose(Function.identity());
     }
 
+    @Nonnull
+    CompletableFuture<Void> gracefullyTerminateOrCancel() {
```

Comment:
`gracefullyTerminate` is still used, so it might cause jobs to remain running (not restarted with new topology)

Reply:
Makes sense, okay.